sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -2056,11 +2056,24 @@ class GpuConvMeta(
   override def tagExprForGpu(): Unit = {
     val fromBaseLit = GpuOverrides.extractLit(expr.fromBaseExpr)
     val toBaseLit = GpuOverrides.extractLit(expr.toBaseExpr)
+    val errorPostfix = "only literal 10 or 16 for from_base and to_base are supported"
     (fromBaseLit, toBaseLit) match {
-      case (Some(Literal(fromBaseVal, IntegerType)), Some(Literal(toBaseVal, IntegerType)))
-        if Set(fromBaseVal, toBaseVal).subsetOf(Set(10, 16)) => ()
+      case (Some(Literal(fromBaseVal, IntegerType)), Some(Literal(toBaseVal, IntegerType))) =>
+        val isBaseSupported = (base: Any) => base match {
+          case 10 | 16 => true
+          case _ => false
+        }
+        if (!isBaseSupported(fromBaseVal) || !isBaseSupported(toBaseVal)) {
+          willNotWorkOnGpu(because = s"either ${fromBaseVal} or ${toBaseVal} is not supported; " +
+            s"${errorPostfix}")
```

Comment:
Rather than "either" which may really need to be "both" for some cases, I'd do separate willNotWorkOnGpu for these.  Also `;` is already used to separate willNotWorkOnGpu reasons, so we don't want to use it within a single reason, e.g.:
```scala
  if (!isBaseSupported(fromBaseVal)) {
    willNotWorkOnGpu(s"$fromBaseVal is not a supported source radix, $errorPostfix")
  }
  if (!isBaseSupported(toBaseVal)) {
    willNotWorkOnGpu(s"$toBaseVal is not a supported target radix, $errorPostfix")
  }
```

Reply:
Added more specialized messages for different cases.